transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -153,29 +173,7 @@ protected void run() {
                         pendingWakeup = true;
                     }
                 }
-            }
-
-            completionQueue.process(this);
-
-            // Always call runAllTasks() as it will also fetch the scheduled tasks that are ready.
-            runAllTasks();
-
-            // Once we submitted its safe to clear the iovArray and so be able to re-use it.
-            submissionQueue.submit();
-            clearUsedIovArrays();
-            try {
-                if (isShuttingDown()) {
-                    closeAll();
-                    // Once we submitted its safe to clear the iovArray and so be able to re-use it.
-                    submissionQueue.submit();
-                    clearUsedIovArrays();
-
-                    if (confirmShutdown()) {
-                        break;
-                    }
-                }
-            } catch (Throwable t) {
-                logger.info("Exception error: {}", t);
+                clearUsedIovArrays();
```

Comment:
@1Jo1 but this is right after submitting (the `submitAndWait()` above)? Unless we skipped it because more work was found, but that's just an edge case (we would have remained in the inner loop anyhow).

Looking again I just realized that there is a (pre-existing) bug here though... the `hasTasks()` check should be inside the try/finally, will fix that.

Reply:
> @1Jo1 but this is right after submitting (the submitAndWait() above)? Unless we skipped it because more work was found, but that's just an edge case (we would have remained in the inner loop anyhow).

exactly that's what I thought :)